packages/sitecore-jss/src/graphql-request-client.ts

```diff
@@ -79,6 +94,17 @@ export class GraphQLRequestClient implements GraphQLClient {
     this.debug = clientConfig.debugger || debuggers.http;
   }
 
+  /**
+   * Factory method for creating a GraphQLRequestClient.
+   * @param {Object} config - client configuration options.
+   * @param {string} config.endpoint - endpoint
+   * @param {string} [config.apiKey] - apikey
+   */
+  static createClientFactory({ endpoint, apiKey }: GraphQLRequestClientFactoryConfig) {
```

Comment:
Is the return type here `GraphQLRequestClientFactory`?
```suggestion
  static createClientFactory({ endpoint, apiKey }: GraphQLRequestClientFactoryConfig): GraphQLRequestClientFactory {
```